zentral/contrib/osquery/serializers.py

```diff
@@ -152,3 +153,46 @@ def iter_query_defaults(self, pack_slug):
                 "compliance_check": query_data.get("compliance_check") or False
             }
             yield query_slug, pack_query_defaults, query_defaults
+
+
+class QuerySerializer(serializers.ModelSerializer):
+    compliance_check_enabled = serializers.BooleanField(default=False)
+
+    class Meta:
+        model = Query
+        exclude = ("compliance_check",)
+
+    def validate(self, data):
+        if data.get("compliance_check_enabled"):
+            sql = data.get("sql")
+            if sql is not None and 'ztl_status' not in sql:
+                raise serializers.ValidationError('{compliance_check_enabled: true} '
```

Comment:
There may be a cleaner way to report the error on the field from the object validation:

https://www.django-rest-framework.org/api-guide/exceptions/#validationerror

> The detail argument may be a list or dictionary of error details, and may also be a nested data structure. By using a dictionary, you can specify field-level errors while performing object-level validation in the validate() method of a serializer. For example. `raise serializers.ValidationError({'name': 'Please enter a valid name.'})`